src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -421,5 +465,189 @@ static AnnotationManager ofAllMinusExcluded(final List<String> annotationsToExcl
+                    if (expressionVC.hasID()) {
+                        attributes.put(expression.fullName, expressionVC.getID());
+                    }
+                } else if (expression.fieldName.equals("ALT")) {
+                    attributes.put(expression.fullName, expressionVC.getAlternateAllele(0).getDisplayString());
+                } else if (expression.fieldName.equals("FILTER")) {
+                    if (expressionVC.isFiltered()) {
+                        attributes.put(expression.fullName, expressionVC.getFilters().toString().replace("[", "").replace("]", "").replace(" ", ""));
+                    } else {
+                        attributes.put(expression.fullName, "PASS");
+                    }
+                } else if (expressionVC.hasAttribute(expression.fieldName)) {
+                    // find the info field
+                    final VCFInfoHeaderLine hInfo = expression.hInfo;
+                    if (hInfo == null) {
+                        throw new UserException("Cannot annotate expression " + expression.fullName + " at " + ref.getInterval() + " for variant allele(s) " + vc.getAlleles() + ", missing header info");
+                    }
+
+                    //
+                    // Add the info field annotations
+                    //
+                    final boolean useRefAndAltAlleles = VCFHeaderLineCount.R == hInfo.getCountType();
+                    final boolean useAltAlleles = VCFHeaderLineCount.A == hInfo.getCountType();
+
+                    // Annotation uses ref and/or alt alleles or enforce allele concordance
+                    if ((useAltAlleles || useRefAndAltAlleles) || expressionAlleleConcordance) {
+
+                        // remove brackets and spaces from expression value
+                        final String cleanedExpressionValue = expressionVC.getAttribute(expression.fieldName,"").toString().replaceAll("[\\[\\]\\s]", "");
+
+                        // get comma separated expression values
+                        final ArrayList<String> expressionValuesList = new ArrayList<>(Arrays.asList(cleanedExpressionValue.split(",")));
+
+                        boolean canAnnotate = false;
+                        // get the minimum biallelics without genotypes
+
+                        final List<VariantContext> minBiallelicVCs = getMinRepresentationBiallelics(vc);
+                        final List<VariantContext> minBiallelicExprVCs = getMinRepresentationBiallelics(expressionVC);
+
+                        // check concordance
+                        final List<String> annotationValues = new ArrayList<>();
+                        for (final VariantContext biallelicVC : minBiallelicVCs) {
+                            // check that ref and alt alleles are the same
+                            List<Allele> exprAlleles = biallelicVC.getAlleles();
+                            boolean isAlleleConcordant = false;
+                            int i = 0;
```

Comment:
There has to be a better way.  This variable can be incremented in two different places, and it is related to the for loop without being syntactically tied to it, and you only need it for the check `i == 0`.